lwip/src/core/mem.c

```diff
@@ -151,7 +152,7 @@ mem_overflow_init_raw(void *p, size_t size)
 }
 #endif /* MEM_OVERFLOW_CHECK || MEMP_OVERFLOW_CHECK */
 
-#if MEM_LIBC_MALLOC || MEM_USE_POOLS
+#if MEM_CUSTOM_ALLOCATOR || MEM_USE_POOLS
```

Comment:
Care needs to be taken with these definition changes, they may affect the performance or behavior of the program.
This is why I prefer to use the release version. You can find any incompatible changes in the changelog.